Support setting the new relic transaction name

rmosolgo opened the pull request:
This adds an option to new relic tracer to set the transaction name for all queries.

akshah123 commented on lib/graphql/tracing/new_relic_tracing.rb:
```diff
@@ -14,7 +14,29 @@ class NewRelicTracing < PlatformTracing
+      #   This is not advised if you run more than one query per HTTP request, for example, with `graphql-client` or multiplexing.
+      def initialize(set_transaction_name: false)
+
+        @set_transaction_name = true
```
looks like this might always set it to `true`.

rmosolgo replied:
🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️

rmosolgo commented on lib/graphql/tracing/new_relic_tracing.rb:
```diff
@@ -14,7 +14,29 @@ class NewRelicTracing < PlatformTracing
+      end
+
       def platform_trace(platform_key, key, data)
+        if @set_transaction_name && platform_key == "execute_query"
```
yeah :S I've been putting off proper unit tests for these platform tracers, but two fatal errors in a row means I can't afford to skip them anymore 😆  thanks!

rmosolgo commented:
Thanks again for the reviews, I've pushed some proper tests and fixed a few issues, and added a simple per-query override. @obrie , thanks for sharing your advanced use case -- I'd like to publish these small improvements as-is, and consider that more sophisticated implementation later, if more folks express interest. It sounds like nice work!
